beams/beams/doctype/compensatory_leave_log/compensatory_leave_log.py

```diff
@@ -1,9 +1,80 @@
-# Copyright (c) 2024, efeone and contributors
-# For license information, please see license.txt
-
-# import frappe
+from datetime import timedelta
+import frappe
+from frappe.utils import getdate, nowdate
 from frappe.model.document import Document
 
 
 class CompensatoryLeaveLog(Document):
-	pass
+    def validate(self):
+        '''Validate the compensatory leave dates'''
+        if self.start_date and self.end_date and getdate(self.end_date) < getdate(self.start_date):
+            frappe.throw('End Date cannot be before Start Date')
+
+@frappe.whitelist()
+def process_expired_compensatory_leaves(self=None, method=None):
+    '''This method check list of CLL and allocated leaves and leave applications and reduce
+        allocated leaves if it is used or expires if unused.'''
+
+    thirty_days_ago = getdate(nowdate()) - timedelta(days=30)
```

Comment:
This is overcomplicating it; getdate() will return the current date in datetime format if no argument is given.

use:
getdate() - timedelta(days=30)

or:
add_days(getdate(), -30)